### PR TITLE
editoast: add roles check to `views::infra`

### PR DESCRIPTION
**Pull request by hamz2a**
- [x] `/infra` GET
- [x] `/infra` POST
- [x] `/infra/refresh`
- [x] `/infra/voltages`
- [x] `/infra/{infra_id}` GET
- [x] `/infra/{infra_id}` DELETE
- [x] `/infra/{infra_id}` PUT
- [x] `/infra/load` 
- [x] `/infra/clone` 
- [x] `/infra/lock` 
- [x] `/infra/unlock` 
- [x] `/infra/speed_limit_tags` 
- [x] `/infra/voltages` 
- [x] `/infra/switch_types`

### CODE REVIEW

File: editoast/editoast_authz/src/builtin_role.rs [hunk 1]
```diff
@@ -5,14 +5,22 @@ use crate::roles::BuiltinRoleSet;
...
+    #[strum(serialize = "infra:write")]
+    InfraWrite,
 }
 
 impl BuiltinRoleSet for BuiltinRole {
     fn implies_iter(&self) -> impl IntoIterator<Item = Self> {
+        use BuiltinRole::*;
         match self {
-            Self::OpsWrite => vec![],
+            OpsWrite => vec![],
```
Comment (flomonster) [inline editoast/editoast_authz/src/builtin_role.rs]: I think we must be able to read infra to write operational studies. 
I'm not 100% sure, what do you think?

```suggestion
            OpsWrite => vec![InfraRead],
```

File: editoast/src/views/infra/mod.rs [hunk 2]
```diff
@@ -119,8 +123,17 @@ struct RefreshResponse {
 )]
 async fn refresh(
     app_state: State<AppState>,
+    Extension(authorizer): AuthorizerExt,
     Query(query_params): Query<RefreshQueryParams>,
 ) -> Result<Json<RefreshResponse>> {
+    let authorized = authorizer
+        .check_roles([BuiltinRole::InfraWrite].into())
```
Comment (flomonster) [inline editoast/src/views/infra/mod.rs:129]: The frontend never uses this. This endpoint is used to force refresh. It's exclusively used by developers (ops)

**Review by flomonster — APPROVED** [review]
LGTM

File: editoast/src/views/infra/mod.rs (hunk 2 above)
Comment (hamz2a) [reply editoast/src/views/infra/mod.rs:130]: ok, Do you want me to delete the check role?

Comment (flomonster) [reply editoast/src/views/infra/mod.rs:129]: Ideally, this endpoint should be available only for superusers.

Comment (hamz2a) [reply editoast/src/views/infra/mod.rs:129]: As discussed, we'll keep the check for now.

File: editoast/editoast_authz/src/builtin_role.rs (hunk 1 above)
Comment (hamz2a) [reply editoast/editoast_authz/src/builtin_role.rs]: It sounds logical.

**Pull request merged**